src/components/OSCALProfile.test.js

```diff
@@ -49,8 +68,54 @@ function testOSCALProfile(parentElementName, renderer) {
   });
 }
 
-testOSCALMetadata("OSCALProfile", profileRenderer);
+function testExternalOSCALProfile(parentElementName, renderer, externalUrl) {
+  test(`${parentElementName} displays controls`, async () => {
+    act(() => {
+      renderer(externalUrl);
+    });
+    const result = await screen.findByText("ca-1", { timeout: 10000 });
+    expect(result).toBeVisible();
+  });
+
+  test(`${parentElementName} displays control modifications`, async () => {
+    renderer(externalUrl);
+    const modButton = await screen.findByRole(
+      "button",
+      { name: "ca-3 modifications" },
+      { timeout: 10000 }
+    );
+    fireEvent.click(modButton);
+    expect(await screen.findByText("Modifications")).toBeVisible();
+    expect(await screen.findByText("Adds")).toBeVisible();
+  });
+}
 
-testOSCALProfile("OSCALProfile", profileRenderer);
+function testUrlOSCALProfile(url) {
+  const externalUrl = url.searchParams.get("url");
+  if (!externalUrl) {
```

Comment:
Could this be done in one line?

Reply:
I don't think so, as `externalUrl` is used a few lines down in calling the `testExternalProfile` functions and `externalUrl` is used in the renderer in the tests. This just makes it so I don't have to call `url.searchParams.get(url)` multiple times.